autoscaler/controllers/gateway/deployment.go

```diff
@@ -25,6 +26,20 @@ const (
 	containerCommand     = "/odigosotelcol"
 	confDir              = "/conf"
 	configHashAnnotation = "odigos.io/config-hash"
+	requestMemory        = "500Mi"
+
+	// this configures the processor limit_mib, which is the hard limit in MiB, afterwhich garbage collection will be forced.
+	// as recommended by the processor docs, this is set to 50MiB less than the memory limit of the collector
+	memoryLimiterLimitMib = 450
+
+	// confgiures the processor spike_limit_mib. When memory usage exceeds the hard limit minus this amount,
+	// the processor will enter memory limited mode and will start refusing data.
+	// as recommended by the processor docs, this is set to 20% of the hard limit
+	memoryLimiterSpikeLimitMiB = 90
```

Comment:
Maybe worth considering defining these constants in a relative way like `0.2 * memoryLimiterLimitMib`. This way we can define a single default memory size of `requestMemory` and calculate all the others based on that value.
That way we could add code that retries with a smaller `requestMemory` in case of failure to deploy (and all the other sizes will adjust)

Reply:
Thanks @RonFed ,

I refactored the code so now it's configurable and relative. Can you please take a look again?